File: src/main/java/com/nugurang/graphql/resolver/UserReviewResolver.java
package com.nugurang.graphql.resolver;

import com.nugurang.dao.UserReviewDao;
import com.nugurang.dto.PositionDto;
import com.nugurang.dto.UserDto;
import com.nugurang.dto.UserEvaluationDto;
import com.nugurang.dto.UserReviewDto;
import graphql.kickstart.tools.GraphQLResolver;
import org.springframework.stereotype.Service;

@Service
public class UserReviewResolver implements GraphQLResolver<UserReviewDto> {
    private final UserReviewDao userReviewDao;

    public PositionDto position(UserReviewDto userReviewDto) {
        return userReviewDao.findById(userReviewDto.getId()).get().getPosition().toDto();
    }

    public UserDto fromUser(UserReviewDto userReviewDto) {
        return userReviewDao.findById(userReviewDto.getId()).get().getFromUser().toDto();
    }

    public UserDto toUser(UserReviewDto userReviewDto) {
        return userReviewDao.findById(userReviewDto.getId()).get().getToUser().toDto();
    }

    public UserEvaluationDto evaluation(UserReviewDto userReviewDto) {
        return userReviewDao.findById(userReviewDto.getId()).get().getUserEvaluation().toDto();
    }

    //<editor-fold defaultstate="collapsed" desc="delombok">
    @SuppressWarnings("all")
    
    public UserReviewResolver(final UserReviewDao userReviewDao) {
        this.userReviewDao = userReviewDao;
    }
    //</editor-fold>
}

File: src/main/java/com/nugurang/graphql/resolver/UserReviewResolver.kt
package com.nugurang.graphql.resolver

import com.nugurang.dao.UserReviewDao
import com.nugurang.dto.PositionDto
import com.nugurang.dto.UserDto
import com.nugurang.dto.UserEvaluationDto
import com.nugurang.dto.UserReviewDto
import com.nugurang.entity.PositionEntity
import com.nugurang.entity.UserEntity
import com.nugurang.exception.NotFoundException
import graphql.kickstart.tools.GraphQLResolver
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service

@Service
class UserReviewResolver(private val userReviewDao: UserReviewDao) : GraphQLResolver<UserReviewDto?> {

    fun position(userReviewDto: UserReviewDto): PositionDto {
        val positionEntity = userReviewDao.findByIdOrNull(userReviewDto.id)?.position
            ?: throw NotFoundException(PositionEntity::class.java)
        return positionEntity.toDto()
    }

    fun fromUser(userReviewDto: UserReviewDto): UserDto {
        val userEntity = userReviewDao.findByIdOrNull(userReviewDto.id)?.fromUser
            ?: throw NotFoundException(UserEntity::class.java)
        return userEntity.toDto()
    }

    fun toUser(userReviewDto: UserReviewDto): UserDto {
        val userEntity = userReviewDao.findByIdOrNull(userReviewDto.id)?.toUser
            ?: throw NotFoundException(UserEntity::class.java)
        return userEntity.toDto()
    }

    fun evaluation(userReviewDto: UserReviewDto): UserEvaluationDto {
        val userEvaluationEntity = userReviewDao.findByIdOrNull(userReviewDto.id)?.userEvaluation
            ?: throw NotFoundException(UserEntity::class.java)
        return userEvaluationEntity.toDto()
    }
}